### PR TITLE
DIGITAL-144: Add and configure Tome, S3, and Samlauth

### PR DESCRIPTION
**Pull request by mattsqd**
## Jira ticket

[DIGITAL-144](https://cm-jira.usa.gov/browse/DIGITAL-144)

## Deployment and testings

We will need to integrate with the deployment PR which will enable / disable the proper splits.

### Local Setup

lando rebuild -y && lando si

### QA/Testing instructions

settings.lando.php will enable you to turn on the non-prod split, which will enable s3 fuse.
The other splits, if enabled, will turn on Samlauth.
You must `./drush.sh cim` after enabling a split.
Tome is always on. You can export the site with `./drush.sh tome:static`

## Checklist for the Developer

- [x] A link to the JIRA ticket has been included above.
- [x] No merge conflicts exist with the target branch.
- [ ] Automated tests have passed on this PR.
- [x] A reviewer has been designated.
- [x] Deployment and testing steps have been documented above, if applicable.

## Checklist for the Peer Reviewers

- [ ] The file changes are relevant to the task objective.
- [ ] Code is readable and includes appropriate commenting.
- [ ] Code standards and best practices are followed.
- [ ] QA/Test steps were successfully completed, if applicable.
- [ ] Applicable logs are free of errors.

### CODE REVIEW

**Comment by mattsqd** [comment]
SAML will be configured later.
s3fuse will be configured via settings overrides by @casey-rapnicki-bixal

**Comment by jlfranklin** [comment]
Let's leave this one open until we have all of the config for SAML Auth.

**Comment by mattsqd** [comment]
> Let's leave this one open until we have all of the config for SAML Auth.



> Let's leave this one open until we have all of the config for SAML Auth.

We need the other configuration in here right now so we can test tome and s3 integration. The Samlauth should not hurt anything if it's not configured right now.

File: config/sync/config_split.config_split.develop.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+dependencies: {  }
+id: develop
+label: Develop
+description: 'Configuration just for the develop branch / dev environment.'
+weight: 0
+stackable: false
+no_patching: false
+storage: folder
+folder: ../config/develop
+module:
```
Comment (jlfranklin) [inline config/sync/config_split.config_split.develop.yml:13]: SAMLAuth and External Auth should be enabled in all environments from prod down to local developer's laptops.  The signing keys and some config will vary, but everyone will use some kind of SAML auth to authenticate in all environments.

File: config/sync/config_split.config_split.non_local.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+id: non_local
+label: Non-Local
+description: 'Non-local environments, ie, in Cloud.gov.'
+weight: 0
+stackable: false
+no_patching: false
+storage: folder
+folder: ../config/non_local
+module:
+  s3fs: 0
```
Comment (jlfranklin) [inline config/sync/config_split.config_split.non_local.yml:14]: Is there an s3-mock container people can use in the locals?

**Review by jlfranklin — CHANGES_REQUESTED** [review]

File: config/sync/config_split.config_split.non_local.yml (hunk 2 above)
Comment (mattsqd) [reply config/sync/config_split.config_split.non_local.yml:14]: Split off to https://cm-jira.usa.gov/browse/DIGITAL-181 as we discussed.

**Comment by mattsqd** [comment]
@jlfranklin ready to review again.

File: config/sync/config_split.config_split.develop.yml (hunk 1 above)
Comment (jlfranklin) [reply config/sync/config_split.config_split.develop.yml]: This works, though adding samlauth to `core.extensions.yml` would be less confusing.

**Review by jlfranklin — APPROVED** [review]

**Pull request merged**